libyul/backends/wasm/EVMToEWasmTranslator.cpp

```diff
@@ -355,8 +791,9 @@ function address() -> z1, z2, z3, z4 {
 	z1, z2, z3, z4 := mload_internal(0)
 }
 function balance(x1, x2, x3, x4) -> z1, z2, z3, z4 {
-	// TODO implement
-	unreachable()
+	mstore_internal(0, x1, x2, x3, x4)
+	eth.getExternalBalance(0, 32)
```

Comment:
I think we need to test that against an actual implementation like hera, but I think, `eth.getExternalBalance` takes an address (from memory) and returns a `u128`. This means, it only reads 20 bytes starting from 0, and only writes 16 bytes starting from 32. So maybe this should rather be
```suggestion
	eth.getExternalBalance(12, 48)
```
plus writing zeros to 32 prior to the call. Furthermore, we might want to check that the input (x1, ..., x4) does not have any higher order bits set...

Reply:
Agreed. Will add higher bits checking.

Reply:
Can we just after calling mload_internal just set z1 and z2 to zeros instead of writing zeros to memory?